### PR TITLE
Move some docs from top-level to admin/, user-guide/, and devel/.

### PR DESCRIPTION
**Pull request by davidopp**
The ones that remain fell into one of two categories
(1) Should not exist on their own -- should be merged into other docs and then deleted (kube-apiserver.md, kube-controller-manager.md, kubelet.md, services-firewall.md)
(2) Geuinely feel like they belong at top-level (api.md, glossary.md, roadmap.md, troubleshooting.md)

I will merge docs/devel/developer-guide.md and docs/devel/README.md in a followup PR.

(This is a redo of #11272, which was clobbered by #11191)

@bgrant0607 @erictune @lavalamp

### CODE REVIEW

**Comment by bgrant0607** [comment]
Got merge conflict. Please rebase.

File: docs/devel/developer-guide.md [hunk 1]
```diff
@@ -24,15 +24,15 @@ certainly want the docs that go with that version.</h1>
 
 The developer guide is for anyone wanting to either write code which directly accesses the
 kubernetes API, or to contribute directly to the kubernetes project.
-It assumes some familiarity with concepts in the [User Guide](user-guide/user-guide.md) and the [Cluster Admin
-Guide](admin/README.md).
+It assumes some familiarity with concepts in the [User Guide](../user-guide/user-guide.md) and the [Cluster Admin
```
Comment (bgrant0607) [inline docs/devel/developer-guide.md]: user-guide.md is gone. Link to user-guide/README.md instead

Comment (bgrant0607) [inline docs/devel/developer-guide.md]: Actually, you should get that change when you rebase

**Comment by bgrant0607** [comment]
Looks good overall.

**Comment by davidopp** [comment]
Rebasing made a mess. I'm going to re-do this PR again. Third time is a charm, maybe? It would be great if we could hold PRs until I can get the new version merged.

**Pull request closed without merging**